Ry0 support

mmpagani opened the pull request:
This PR adds support for ry0 in the calculation of the distance context i.e. an instance of :class:`DistancesContext`. Test are green https://ci.openquake.org/job/zdevel_oq-hazardlib/179/

g-weatherill commented:
This PR seems to implement all the features of #269, plus the addition of the DistancesContext. The code LGTM, so I think the best thing to do is to close PR #269 and merge this one, once the developers approve it.

micheles commented on openquake/hazardlib/geo/surface/base.py:
```diff
@@ -225,6 +239,79 @@ def get_joyner_boore_distance(self, mesh):
+        dst = numpy.fmin(numpy.abs(dst1), numpy.abs(dst2))
+
+        return dst
+
```
Usually in such cases one define two methods, say get_ry0_distance_option_1 and get_ry0_distance_option_2 and call them inside get_ry0_distance, inside of having a big `if`.

mmpagani replied:
I removed the method in the option 2. This way the implementation is cleaner.

micheles commented on openquake/hazardlib/tests/geo/surface/base_surface_test.py:
```diff
@@ -117,6 +120,31 @@ def test_point_outside(self):
+        expected_dists = [111.19505230826488, 111.177990689, 111.177990689,
+                          111.19505230826488, 111.177990689, 111.177990689]
+        self.assertTrue(numpy.allclose(dists, expected_dists))
+
```
I don't see the test for the case `option 2`

mmpagani commented:
I think I addressed the comments of the reviewers. Tests for the updated version are green
https://ci.openquake.org/job/zdevel_oq-hazardlib/192/?auto_refresh=false

micheles commented:
LGTM

g-weatherill commented:
LGTM too
